src/parse.rs

```diff
@@ -0,0 +1,319 @@
+                        Some(Parse::Complete(Node::Transclusion(trans)))
+                    }
+                }
+            }
+        })
+        .filter_map(|parse| match parse {
+            Parse::Incomplete => None,
+            Parse::Error(error) => Some(Err(error)),
+            Parse::Complete(node) => Some(Ok(node)),
+        })
+        .try_collect()
+        .map_err(|errors| {
+            let mut msg = String::new();
+            for error in errors {
+                writeln!(&mut msg, "{}", error).unwrap();
+            }
+            msg
+        })?;
+    if let Some(node) = state.node.take() {
+        document.push(node);
+    }
+    Ok(Document::new(document))
+}
+
+fn parse_transclusion(
+    line: &str,
+    into: &Path,
+    settings: &ParserSettings,
+) -> Fallible<Option<Node>> {
+    if let Some(rest) = line.trim().strip_prefix(&settings.transclusion_start) {
+        if let Some(index) = rest.find(&settings.transclusion_end) {
+            let trans = &rest[..index];
+
+            let target = LINK_REGEX
+                .captures_iter(trans)
+                .map(|match_| match_.get(2).unwrap().as_str())
+                .next()
+                .unwrap_or(&trans);
+
+            let mut path = PathBuf::from(into.parent().unwrap_or_else(|| Path::new(".")));
+            path.push(target);
+
+            Ok(Some(Node::Transclusion(Transclusion::new(
+                path,
+                line.to_owned(),
+            ))))
+        } else {
+            Err(format!("Unclosed transclusion in: {}", line).into())
+        }
+    } else {
+        Ok(None)
+    }
+}
+
+/// Parses a line as code, returning the parsed `Line` object
+fn parse_line(line_number: usize, input: &str, settings: &ParserSettings) -> Fallible<Line> {
+    let indent_len = input
+        .chars()
+        .take_while(|ch| ch.is_whitespace())
+        .collect::<String>()
+        .len();
+    let (indent, rest) = input.split_at(indent_len);
+
+    // TODO: Temporarily disables comment extraction.
+    let (rest, comment) = (rest, None);
+    /*let (rest, comment) = if let Some(comment_index) = rest.find(&settings.block_name_prefix) {
+        let (rest, comment) = rest.split_at(comment_index);
+        (
+            rest,
+            Some((&comment[settings.block_name_prefix.len()..]).to_owned()),
+        )
+    } else {
+        (rest, None)
+    };*/
+
+    if let Some(stripped) = rest.strip_prefix(&settings.macro_start) {
+        if let Some(name) = stripped.strip_suffix(&settings.macro_end) {
+            return Ok(Line {
+                line_number,
+                indent: indent.to_owned(),
+                source: Source::Macro(name.trim().to_owned()),
+                comment,
+            });
+        }
+    }
+
+    Ok(Line {
+        line_number,
+        indent: indent.to_owned(),
+        source: Source::Source(rest.to_owned()),
+        comment,
+    })
+}
+
+pub fn find_links(
```

Comment:
This seems to conflate two things: Finding (and returning) the links in the document and removing the markers which would probably best happen already during parsing or at least as a separate post-processing step?

Reply:
This is probably best combined with https://github.com/mlange-42/yarner/pull/89#discussion_r571616614 and https://github.com/mlange-42/yarner/pull/89#discussion_r571617366 into a separate MR. See #97.